src/coreclr/jit/lsrabuild.cpp

```diff
@@ -1936,6 +1936,10 @@ void LinearScan::buildPhysRegRecords()
         RegRecord* curr = &physRegs[reg];
         curr->init(reg);
     }
+#if defined(TARGET_AMD64)
```

Comment:
Doesn't this fall out naturally? Or are there cases where `REG_NEXT(get_REG_INT_LAST()) != REG_FP_FIRST`?

Should this be all-platform (no ifdef)?

Reply:
There are 2 cases
 1. `x64` with `APX ON`  
 `get_REG_INT_LAST()` = `REG_R31` and so` REG_NEXT(get_REG_INT_LAST()) = REG_FP_FIRST`
 2. `x64` with `APX OFF` 
 `get_REG_INT_LAST()` = `REG_R15` and so` REG_NEXT(get_REG_INT_LAST()) != REG_FP_FIRST`
This is because `REG_NEXT(get_REG_INT_LAST())` = `REG_R16`. `REG_R16` is defined but not valid for `x64` targets with no APX support